src/Symfony/Bridge/Doctrine/Form/ChoiceList/IdReader.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\Form\ChoiceList;
```

Comment:
What about moving the class out of the `Form` namespace? The `UniqueEntityConstraintValidator` and the `EntityUserProvider` use some similar logic to retrieve id values for entities.